proto/storage/traits.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+option go_package = "storage";
+option java_package = "io.stackrox.proto.storage";
+
+import "gogoproto/gogo.proto";
+
+package storage;
+
+// EXPERIMENTAL.
+//
+// MutabilityMode specifies whether and how an object can be modified. Default
+// is ALLOW and means there are no modification restrictions; this is equivalent
+// to the absence of MutabilityMode specification. ALLOW_FORCED forbids all
+// modifying operations except object removal with force bit on.
+//
+// Be careful when changing the state of this field. For example, modifying an
+// object from ALLOW to ALLOW_FORCED is allowed but will prohibit any further
+// changes to it, including modifying it back to ALLOW.
+enum MutabilityMode {
```

Comment:
Please move this within `Traits`. Proto enums follow C scoping rules, which means `ALLOW` is visible at the `storage` package's top-level namespace. Additionally or alternatively, it might be good to call this something like `MUTABILITY_ALLOW`, `ALLOW_MUTATE` etc., as even within `Traits` there may be more use cases for `ALLOW`ing something.